src/test/java/picard/fingerprint/CrosscheckReadGroupFingerprintsTest.java

```diff
@@ -0,0 +1,312 @@
+package picard.fingerprint;
+
+import htsjdk.samtools.metrics.MetricsFile;
+import org.testng.Assert;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import picard.vcf.SamTestUtils;
+
+import java.io.*;
+import java.lang.reflect.Field;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.util.*;
+import java.util.stream.Stream;
+
+import static picard.fingerprint.FingerprintIdDetails.multipleValuesString;
+
+/**
+ * Tests for CrosscheckReadGroupFingerprints
+ */
+public class CrosscheckReadGroupFingerprintsTest {
+
+    private final static File TEST_DIR = new File("testdata/picard/fingerprint/");
+    private final static File HAPLOTYPE_MAP = new File(TEST_DIR, "Homo_sapiens_assembly19.haplotype_database.subset.txt");
+
+    static private final File NA12891_r1_sam = new File(TEST_DIR, "NA12891.over.fingerprints.r1.sam");
+    static private final File NA12891_r2_sam = new File(TEST_DIR, "NA12891.over.fingerprints.r2.sam");
+
+    //this is a copy of a previous one, but with a different sample name
+    static private final File NA12891_named_NA12892_r1_sam = new File(TEST_DIR, "NA12891_named_NA12892.over.fingerprints.r1.sam");
+
+    static private final File NA12892_r1_sam = new File(TEST_DIR, "NA12892.over.fingerprints.r1.sam");
+    static private final File NA12892_r2_sam = new File(TEST_DIR, "NA12892.over.fingerprints.r2.sam");
+
+    static private File NA12891_r1, NA12891_r2, NA12891_named_NA12892_r1, NA12892_r1, NA12892_r2;
+
+    static private final int NA12891_r1_RGs = 27;
+    static private final int NA12891_r2_RGs = 26;
+    static private final int NA12892_r1_RGs = 25;
+    static private final int NA12892_r2_RGs = 26;
+
+    private static final Map<CrosscheckMetric.DataType, List<String>> lookupMap = new HashMap<>(4);
+
+    @BeforeTest
+    static public void setup() throws IOException {
+        NA12891_r1 = SamTestUtils.createIndexedBam(NA12891_r1_sam, NA12891_r1_sam);
+        NA12891_r2 = SamTestUtils.createIndexedBam(NA12891_r2_sam, NA12891_r2_sam);
+        NA12891_named_NA12892_r1 = SamTestUtils.createIndexedBam(NA12891_named_NA12892_r1_sam, NA12891_named_NA12892_r1_sam);
+        NA12892_r1 = SamTestUtils.createIndexedBam(NA12892_r1_sam, NA12892_r1_sam);
+        NA12892_r2 = SamTestUtils.createIndexedBam(NA12892_r2_sam, NA12892_r2_sam);
+
+        lookupMap.put(CrosscheckMetric.DataType.FILE, new ArrayList<>());
+        lookupMap.get(CrosscheckMetric.DataType.FILE).addAll(Arrays.asList("LEFT_FILE", "RIGHT_FILE"));
```

Comment:
You don't need to do this in two steps, you could write
```
lookupMap.put(CrosscheckMetric.DataType.FILE, new ArrayList<>(Arrays.asList("LEFT_FILE", "RIGHT_FILE")));
```

If you don't need the list to be modifiable then you can skip the `new ArrayList<>()` constructor call.